lib/config.js

```diff
@@ -463,21 +463,21 @@ function normalizeTime (opts) {
   }
 }
 
-function maybeSplit (separator) {
-  return (value) => {
-    return typeof value === 'string' ? value.split(separator) : value
-  }
-}
-
-const maybeSplitValues = maybeSplit(',')
-const maybeSplitPairs = maybeSplit('=')
-
+// Array config vars are either already an array of strings, or a
+// comma-separated string (whitespace is trimmed):
+//    'foo, bar' => ['foo', 'bar']
 function normalizeArrays (opts) {
   for (const key of ARRAY_OPTS) {
-    if (key in opts) opts[key] = maybeSplitValues(opts[key])
+    if (key in opts && typeof opts[key] === 'string') {
+      opts[key] = opts[key].split(',').map(v => v.trim())
+    }
   }
 }
 
+// KeyValuePairs config vars are either an object or a comma-separated string
+// of key=value pairs (whitespace around the "key=value" strings is trimmed):
+//    {'foo': 'bar', 'eggs': 'spam'} => [['foo', 'bar'], ['eggs', 'spam']]
+//    foo=bar, eggs=spam             => [['foo', 'bar'], ['eggs', 'spam']]
 function normalizeKeyValuePairs (opts) {
   for (const key of KEY_VALUE_OPTS) {
```

Comment:
It looks like we're refactoring `normalizeKeyValuePairs` here -- I don't see a lot of test coverage for this method in `lib/config.js` (maybe I'm missing it?).  How would you feel about expanding our test coverage so it's similar in robustness to the _parsing of ARRAY and KEY_VALUE opts_ test?

Reply:
This is tested by these lines: https://github.com/elastic/apm-agent-nodejs/pull/1914/files#diff-324e5ae6ed1d64846d943bc35a8a7394f2099af356ca5d9e30413648a3ae124fR923-R938

Is there more that you think should be tested there?

Reply:
I was thinking of of some direct tests of `normalizeKeyValuePairs` -- but the indirect tests you pointed out should be sufficient